mypy/semanal.py

```diff
@@ -3765,6 +3765,10 @@ def analyze_alias(
                     last_tvar_name_with_default = tvar_def.name
                 tvar_defs.append(tvar_def)
 
+            if python_3_12_type_alias:
+                with self.allow_unbound_tvars_set():
+                    rvalue.accept(self)
```

Comment:
This change seems to add a new false-positive for simple type aliases.
```py
type Alias = type[int]
```

```
error: The type "type[type]" is not generic and not indexable  [misc]
```

Reply:
This should have been fixed by #17404. Can you try it out?